S3Scanner/__main__.py

```diff
@@ -181,13 +185,22 @@ def main():
         if args.dangerous:
             print("INFO: Including dangerous checks. WARNING: This may change bucket ACL destructively")
 
+        results = list()
         with ThreadPoolExecutor(max_workers=args.threads) as executor:
             futures = {
-                executor.submit(scan_single_bucket, s3service, anons3service, args.dangerous, bucketName): bucketName for bucketName in bucketsIn
+                executor.submit(scan_single_bucket, s3service, anons3service, args.dangerous, args.json_file, bucketName): bucketName for bucketName in bucketsIn
             }
             for future in as_completed(futures):
                 if future.exception():
                     print(f"Bucket scan raised exception: {futures[future]} - {future.exception()}")
+                elif args.json_file:
+                    results.append(future.result())
+            
+        if args.json_file:
+            with open(f"{args.json_file}", "w") as result_file:
```

Comment:
Instead of creating a list, appending to it, and then writing the list at the end, let's open the file before scanning starts and then write to it as results come in. What we're trying to avoid is a user scanning a large list of buckets, getting most of the way through, and then something happening (like a system shutdown or an app crash) leaving the results file empty because execution didn't get to here yet.